src/wp-admin/css/forms.css

```diff
@@ -782,6 +782,30 @@ ul#add-to-blog-users {
 	outline: 2px solid transparent;
 }
 
+.button-add-site-icon{
+	width: 100%;
+	cursor: pointer;
+	text-align: center;
+	cursor: default;
```

Comment:
The cursor should be `pointer` and it's already set two lines above.

Also important: this button doesn't use the WP focus style. It does use the native browsers focus style, which is okay from an accessibility perspective, but it's inconsistent with the custom focus style provided by WordPress. I'd suggest to add  `button` class to the HTML (that would provide the WP focus style) and then adjust this custom styling accordingly.